src/shared/codelens/localLambdaRunner.ts

```diff
@@ -355,66 +356,85 @@ export const makeBuildDir = async (): Promise<string> => {
     return buildDir
 }
 
-export async function makeInputTemplate(params: {
-    baseBuildDir: string,
-    codeDir: string,
-    documentUri: vscode.Uri
-    originalHandlerName: string,
+export function getHandlerRelativePath(params: {
+    codeRoot: string,
+    filePath: string
+}): string {
+    return path.relative(params.codeRoot, path.dirname(params.filePath))
+}
+
+export function getRelativeFunctionHandler(params: {
     handlerName: string,
     runtime: string,
-    workspaceUri: vscode.Uri,
-}): Promise<string> {
-    const inputTemplatePath: string = path.join(params.baseBuildDir, 'input', 'input-template.yaml')
-    ExtensionDisposableFiles.getInstance().addFolder(inputTemplatePath)
-
+    handlerFileRelativePath: string
+}): string {
     // Make function handler relative to baseDir
-    const handlerFileRelativePath = path.relative(
-        params.codeDir,
-        path.dirname(params.documentUri.fsPath)
-    )
+    let relativeFunctionHandler: string
+    if (isInterpretedRuntime(params.runtime)) {
+        relativeFunctionHandler = normalizeSeparator(
+            path.join(
+                params.handlerFileRelativePath,
+                params.handlerName,
+            )
+        )
+    } else {
+        relativeFunctionHandler = params.handlerName
+    }
 
+    return relativeFunctionHandler
+}
+
+export async function getLambdaInfoFromExistingTemplate(params: {
+    workspaceUri: vscode.Uri,
+    originalHandlerName: string,
+    runtime: string,
+    handlerFileRelativePath: string
+}): Promise <LocalLambda | undefined> {
     const workspaceFolder = vscode.workspace.getWorkspaceFolder(params.workspaceUri)
-    let existingTemplateResource: CloudFormation.Resource | undefined
+    let existingLambda: LocalLambda | undefined
     if (workspaceFolder) {
 
-        const relativeOriginalFunctionHandler = normalizeSeparator(
-            path.join(
-                handlerFileRelativePath,
-                params.originalHandlerName,
-            )
-        )
+        const relativeOriginalFunctionHandler = getRelativeFunctionHandler({
```

Comment:
Calling code already has to call `getRelativeFunctionHandler`. What if we don't calculate it in this method, just take it as a `handler` argument? We would reduce the amount of args in this method as well.

Reply:
It's not the same value. This is a relative function handler based off the original handler and the other one is for the possibly-modified function handler (see: Python). The logic to calculate the function handler is the same, but the base value is different.

Reply:
Thank you for pointing that out, I didn't see that subtle difference. Would it be reasonable for callers like Python to pre-generate this handler? It already has to be explicit in providing that specific handler.